src/RecurringSwap.sol

```diff
@@ -93,20 +102,29 @@ contract RecurringSwap is QuarkScript {
         }
 
         bytes32 hashedConfig = _hashConfig(config);
-        uint256 nextSwapTime;
+        uint256 nextWindowStart;
         if (read(hashedConfig) == 0) {
-            nextSwapTime = config.startTime;
+            nextWindowStart = config.swapWindow.startTime;
         } else {
-            nextSwapTime = uint256(read(hashedConfig));
+            nextWindowStart = uint256(read(hashedConfig));
```

Comment:
It might just be prudent to say this is "completedIntervals" or "currentIntervalNumber" since it's a _bit_ more confusing to store a timestamp versus a simple small integer.

Reply:
E.g. then a lot of the code below is basically "is currentInterval > lastInterval" and then "block.timestamp > currentInterval * intervalLength && block.timestamp <= currentInterval * intervalLength + windowLength", which is pretty close to what you have, I guess

Reply:
Yeah, that could also work. I'll play around with this alternative implementation and start a new PR if it looks good.